### ISSUE TITLE
URI module fails with unicode error when downloading zipped file

### ISSUE THREAD

**Issue by jvanns** [ansible/ansible-modules-core#2088]
OS: Ubuntu 14.04
Python: 2.7.6 (stock install from distribution)
Ansible: 1.9.3 (installed via pip)
Module: URI

Reproduce with: 

```
ansible localhost -m uri -a "url='http://dl.bintray.com/mitchellh/consul/0.5.2_linux_amd64.zip' method=GET creates=/tmp/consul.zip dest=/tmp/consul.zip"
```

Result:

```
localhost | FAILED >> {
    "failed": true, 
    "msg": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1442399748.76-133951194832924/uri\", line 2081, in <module>\n    main()\n  File \"/root/.ansible/tmp/ansible-tmp-1442399748.76-133951194832924/uri\", line 447, in main\n    u_content = unicode(content, content_encoding, errors='xmlcharrefreplace')\nTypeError: don't know how to handle UnicodeDecodeError in error callback\n", 
    "parsed": false
}
```

HTTP headers (from wget -S):

```
  HTTP/1.1 200 OK
  Content-Type: application/unknown
  Content-Length: 4803609
  Connection: keep-alive
  Date: Sun, 13 Sep 2015 08:20:14 GMT
  Content-Disposition: attachment;filename="0.5.2_linux_amd64.zip"
  Last-Modified: Mon, 18 May 2015 21:36:44 GMT
  ETag: "37000419d608fd34f0f2d97806cf7399"
  Accept-Ranges: bytes
  Server: AmazonS3
  Age: 5862
  X-Cache: Hit from cloudfront
  Via: 1.1 e874817f6498def18bc3c2de8e842383.cloudfront.net (CloudFront)
  X-Amz-Cf-Id: 7J8b-qJl0Dx06c3QG77jBOV_9NB3fz8sGderkwRE8QcHL2x2JWMwkA==
Length: 4803609 (4.6M) [application/unknown]
```

Although the content type maybe unknown, just download the data verbatim? Just like curl or wget ;)

Note we have not 'messed' with Python's default encoding - we use the execution environment as provided by Ubuntu.

**Comment by mikeputnam** [comment]
_BUMP_   A uri module play that worked in 1.9.2 no longer works in 1.9.3 or 1.9.4 citing the exact error above.

Was able to workaround by re-writing the task to use the `get_url` module instead.

**Comment by abadger** [comment]
Fixed in devel 11833906b58edd1542b7e7a54c6f1a5ae78c0d76 and stable-1.9 https://github.com/ansible/ansible-modules-core/commit/73684317ea5b6f9d993256dd9aeff3af05ec10e5  (with an additional commit to get rid of the debugging that was accidentally left in there devel:  https://github.com/ansible/ansible-modules-core/commit/49ee5dd443d81dc3d989ab037ba13313f5215233 stable: https://github.com/ansible/ansible-modules-core/commit/189c741ed1706207365007e4d960f5881269e30e)

# Closing This Ticket

Hi!

We believe recent commits (likely detailed above) should resolve this question or problem for you.

This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!
